homeassistant/components/compit/climate.py

```diff
@@ -0,0 +1,268 @@
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        value = self.get_parameter_value(CompitParameter.CURRENT_TEMPERATURE)
+        if value is None:
+            return None
+        return float(value.value)
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach."""
+        value = self.get_parameter_value(CompitParameter.SET_TARGET_TEMPERATURE)
+        if value is None:
+            return None
+        return float(value.value)
+
+    @cached_property
+    def preset_modes(self) -> list[str] | None:
+        """Return the available preset modes."""
+        if self.available_presets is None or self.available_presets.details is None:
+            return []
+
+        preset_modes = []
+        for item in self.available_presets.details:
+            if item is not None:
+                ha_preset = COMPIT_PRESET_MAP.get(CompitPresetMode(item.state))
+                if ha_preset and ha_preset not in preset_modes:
+                    preset_modes.append(ha_preset)
+
+        return preset_modes
+
+    @cached_property
+    def fan_modes(self) -> list[str] | None:
+        """Return the available fan modes."""
+        if self.available_fan_modes is None or self.available_fan_modes.details is None:
+            return []
+
+        fan_modes = []
+        for item in self.available_fan_modes.details:
+            if item is not None:
+                ha_fan_mode = COMPIT_FANSPEED_MAP.get(CompitFanMode(item.state))
+                if ha_fan_mode and ha_fan_mode not in fan_modes:
+                    fan_modes.append(ha_fan_mode)
+
+        return fan_modes
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode."""
+        preset_mode = self.get_parameter_value(CompitParameter.PRESET_MODE)
+
+        if preset_mode:
+            compit_preset_mode = CompitPresetMode(preset_mode.value)
+            return COMPIT_PRESET_MAP.get(compit_preset_mode)
+        return None
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the current fan mode."""
+        fan_mode = self.get_parameter_value(CompitParameter.FAN_MODE)
+        if fan_mode:
+            compit_fan_mode = CompitFanMode(fan_mode.value)
+            return COMPIT_FANSPEED_MAP.get(compit_fan_mode)
+        return None
+
+    @property
+    def hvac_mode(self) -> HVACMode | None:
+        """Return the current HVAC mode."""
+        hvac_mode = self.get_parameter_value(CompitParameter.HVAC_MODE)
+        if hvac_mode:
+            compit_hvac_mode = CompitHVACMode(hvac_mode.value)
+            return COMPIT_MODE_MAP.get(compit_hvac_mode)
+        return None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        if temp is None:
+            raise ValueError("Temperature argument missing")
```

Comment:
Lets raise `ServiceValidationError` instead 
See https://developers.home-assistant.io/docs/core/platform/raising_exceptions

Reply:
Same for the other service calls